### PR TITLE
feat: add support to show and hide filter search in tables

### PR DESCRIPTION
**Pull request by Nabin99**

### CODE REVIEW

File: packages/user/src/components/Invitation/InvitationsTable.tsx [hunk 1]
```diff
@@ -44,6 +44,7 @@ export type InvitationsTableProperties = {
   prepareInvitationData?: (data: any) => any;
   roles?: Array<InvitationRoleOption>;
   showAppColumn?: boolean;
+  showFilterSearch?: boolean;
```
Comment (suvashvishowkarma) [inline packages/user/src/components/Invitation/InvitationsTable.tsx:47]: @Nabin99 should we use `filterableColumns: FilterableColumn[]` like visibleColumns? what do you think? If we want this way, maybe we can change it to `showEmailFilter`.

File: packages/ui/src/FileCard/index.tsx [hunk 2]
```diff
@@ -152,15 +146,15 @@ export const FileCard = ({
         <div className="file-details-wrapper">
           <div className="file-name-description-details-wrapper">
             <div>
-              {visibleFileDetailsObject.filename ? (
+              {visibleFileDetailsMap.filename ? (
                 <span className="file-name">{file.filename}</span>
               ) : null}
-              {file.size && visibleFileDetailsObject.filesize && (
+              {file.size && visibleFileDetailsMap.filesize && (
```
Comment (suvashvishowkarma) [inline packages/ui/src/FileCard/index.tsx:152]: `filesize` -> `fileSize` typo here?

File: packages/ui/src/utils/UseVisibleOrFilterableField.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { useMemo } from "react";
+
+export const useVisibleOrFilterableField = <T extends string>(
```
Comment (suvashvishowkarma) [inline packages/ui/src/utils/UseVisibleOrFilterableField.ts:3]: Maybe we can change this to more generic name eg `useColumnsMap`? :thinking:

**Review by suvashvishowkarma — COMMENTED** [review]
PR looks good, only a few comments. Also make sure that the filters row (empty) is not displayed when no column is provided in filterableColumns prop.

File: packages/ui/src/FileCard/index.tsx (hunk 2 above)
Comment (Nabin99) [reply packages/ui/src/FileCard/index.tsx:152]: Not actually, field name is written similarly (filename, filesize).

File: packages/ui/src/utils/UseVisibleOrFilterableField.ts (hunk 3 above)
Comment (Nabin99) [reply packages/ui/src/utils/UseVisibleOrFilterableField.ts:3]: sounds good.

File: packages/ui/src/utils/UseColumnsMap.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  return useMemo(() => {
+    const visibleDetails = {} as Record<T, boolean>;
+
+    for (const name of visibleFields) {
+      visibleDetails[name] = true;
+    }
+
+    return visibleDetails;
+  }, [visibleFields]);
+};
```
Comment (suvashvishowkarma) [inline packages/ui/src/utils/UseColumnsMap.ts:13]: update `visibleFields` to `columns` or `fields`

**Pull request merged**